fixed-price-subscriptions/client/react/src/Prices.js

```diff
@@ -1,65 +1,46 @@
 import React, { useState } from 'react';
 import { withRouter } from 'react-router-dom';
-import TopNavigationBar from './TopNavigationBar';
-import StripeSampleFooter from './StripeSampleFooter';
-import PaymentForm from './PaymentForm';
-import Product from './Product';
+import { Redirect } from 'react-router-dom';
 
-const products = [
-  {
-    key: 0,
-    price: '$5.00',
-    name: 'Basic',
-    interval: 'month',
-    billed: 'monthly',
-  },
-  {
-    key: 1,
-    price: '$15.00',
-    name: 'Premium',
-    interval: 'month',
-    billed: 'monthly',
-  },
-];
+const Prices = () => {
+  const [priceLookupKey, setPriceLookupKey] = useState(null);
 
-function Prices({ location }) {
-  const [productSelected, setProduct] = useState(null);
-  const [customer] = useState(location.state.customer);
-
-  function handleClick(key) {
-    setProduct(products[key]);
+  if (priceLookupKey) {
+    return <Redirect to={{
+      pathname: '/subscribe',
+      state: { priceLookupKey }
+    }} />
   }
 
   return (
-    <div className="p-6">
-      <TopNavigationBar />
-      <div className="flex flex-wrap justify-center">
-        <div className="md:w-1/3 w-full mr-4 md:mb-8">
-          <div className="text-center text-pasha font-bold text-2xl mt-4 mb-6">
-            Subscribe to a plan
-          </div>
+    <div>
+      <h1>Select a plan</h1>
+
+      <div className="price-list">
+        <div>
+          <h3>Basic</h3>
 
-          <div className="flex justify-between mb-8">
-            {products.map((product, index) => {
-              return (
-                <Product
-                  key={index}
-                  product={product}
-                  handleClick={handleClick}
-                />
-              );
-            })}
-          </div>
-          {productSelected ? (
-            <PaymentForm
-              productSelected={productSelected}
-              customer={customer}
-            />
-          ) : null}
+          <p>
+            $5.00 / month
+          </p>
+
+          <button onClick={setPriceLookupKey.bind(null, "basic")}>
+            Select
+          </button>
         </div>
-      </div>
 
-      <StripeSampleFooter />
+        <div>
+          <h3>Premium</h3>
+
+          <p>
+            $15.00 / month
+          </p>
+
+          <button onClick={setPriceLookupKey.bind(null, "premium")}>
```

Comment:
This is fine, but a simple anonymous function might be more familiar to folks than `.bind`.

Reply:
I might be a little rusty on React, but I thought that `bind` was preferred over anonymous functions as with anonymous functions a new one gets created on every render, whereas with `bind` you just have the one instance.

Reply:
@paulasjes-stripe both bind and anonymous arrow functions create a new function on each render [1]. In the early days it was thought to be a big deal, but in reality the performance hit is negligible for most applications. Maybe if you're doing animations or canvas renders the new function could make some measurable impact. But, in the majority of apps I would say your better off focusing on the real bottlenecks like client-server comms, and bundle management, whilst trying to make the code as readable as possible.

[1] https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md

Reply:
For a bit of added colour, there's been a more than steady gradual shift away from class components to functional components, where it's very common to define new functions within the functional component knowing full well that they will be newly defined on each render. Instead of discouraging the pattern, knowing full well there aren't many other options in a lot of cases, React introduced memoization utils for cases where re-renders happen constantly and it actually begins to matter (e.g., https://reactjs.org/docs/hooks-reference.html#usecallback)